boot/zephyr/boards/nrf52840_pca10059.conf

```diff
@@ -28,3 +28,5 @@ CONFIG_USB_MASS_STORAGE=n
 
 # Logging
 CONFIG_SYS_LOG_DEFAULT_LEVEL=1
+
+CONFIG_MPU_ALLOW_FLASH_WRITE=y
```

Comment:
Add to Kconfig (here: https://github.com/JuulLabs-OSS/mcuboot/blob/master/boot/zephyr/Kconfig#L12)
`select MPU_ALLOW_FLASH_WRIT if ARM_MPU`
so will be enabled for every ARM-MPU equipped device

Reply:
Done. much cleaner way to fix it. Thanks.